vllmd-hypervisor/generate-runtime-image.sh

```diff
@@ -0,0 +1,593 @@
+        missing=1
+    fi
+    
+    if [[ "${missing}" -eq 1 ]]; then
+        exit 1
+    fi
+}
+
+# Check for existing files
+check_existing_files() {
+    local existing=0
+    
+    if [[ -f "${OUTPUT_PATH}" ]]; then
+        echo "WARNING: Output image already exists at ${OUTPUT_PATH}"
+        existing=1
+    fi
+    
+    if [[ "${existing}" -eq 1 && "${FORCE}" -ne 1 ]]; then
+        echo "Use --force to overwrite existing files"
+        exit 1
+    fi
+}
+
+# Create necessary directories
+create_directories() {
+    if [[ ! -d "${STATE_DIR}" ]]; then
+        if [[ "${DRY_RUN}" -eq 0 ]]; then
+            mkdir -p "${STATE_DIR}"
+            echo "Created state directory: ${STATE_DIR}"
+        else
+            echo "[DRY RUN] Would create state directory: ${STATE_DIR}"
+        fi
+    fi
+    
+    # Create build directory for temporary files with timestamp
+    BUILD_DIR="${STATE_DIR}/${TIMESTAMP}-build"
+    if [[ ! -d "${BUILD_DIR}" ]]; then
+        if [[ "${DRY_RUN}" -eq 0 ]]; then
+            mkdir -p "${BUILD_DIR}"
+            echo "Created build directory: ${BUILD_DIR}"
+        else
+            echo "[DRY RUN] Would create build directory: ${BUILD_DIR}"
+        fi
+    fi
+}
+
+# Download Debian netinst ISO
+download_installer() {
+    local iso_url="https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-12.9.0-amd64-netinst.iso"
+    local iso_path="${BUILD_DIR}/debian-netinst.iso"
+    local cache_iso="${CACHE_DIR}/debian-netinst.iso"
+    
+    # Create cache directory if it doesn't exist
+    if [[ ! -d "${CACHE_DIR}" ]]; then
+        if [[ "${DRY_RUN}" -eq 0 ]]; then
+            mkdir -p "${CACHE_DIR}"
+            echo "Created cache directory: ${CACHE_DIR}"
+        else
+            echo "[DRY RUN] Would create cache directory: ${CACHE_DIR}"
+        fi
+    fi
+    
+    # Check if file exists in cache
+    if [[ -f "${cache_iso}" ]]; then
+        echo "Found cached ISO at ${cache_iso}, reusing..."
+        if [[ "${DRY_RUN}" -eq 0 ]]; then
+            cp "${cache_iso}" "${iso_path}" || { echo "Failed to copy cached ISO"; exit 1; }
+            echo "Copied to ${iso_path}"
+        else
+            echo "[DRY RUN] Would copy cached ISO to ${iso_path}"
+        fi
+        return
+    elif [[ -f "${iso_path}" ]]; then
+        echo "Using existing ISO at ${iso_path}"
+        return
+    fi
+    
+    if [[ "${DRY_RUN}" -eq 0 ]]; then
+        echo "Downloading Debian netinst ISO..."
+        /usr/bin/curl -L -o "${iso_path}" "${iso_url}" || { echo "Failed to download ISO"; exit 1; }
+        echo "Download completed."
+        
+        # Cache the ISO for future runs
+        echo "Caching ISO for future use..."
+        cp "${iso_path}" "${cache_iso}" || { echo "Warning: Failed to cache ISO"; }
+    else
+        echo "[DRY RUN] Would download Debian netinst ISO from:"
+        echo "  - ${iso_url}"
+    fi
+}
+
+# Create preseed configuration disk
+create_preseed_disk() {
```

Comment:
you are creating a preseed image, are you not? You are not creating a disk, cluod-hy0pervisor does that.